smart_open/gcs.py

```diff
@@ -209,11 +209,13 @@ def __init__(
     ):
         if client is None:
             client = google.cloud.storage.Client()
+
+        self._bucket_name = bucket
         bucket = client.get_bucket(bucket)  # type: google.cloud.storage.Bucket
 
         self._blob = bucket.get_blob(key)
         if self._blob is None:
-            raise google.cloud.exceptions.NotFound('blob {} not found in {}'.format(key, bucket))
+            raise google.cloud.exceptions.NotFound('blob {} not found in {}'.format(key, self._bucket_name))
```

Comment:
Unrelated to your PR but we avoid `.format()` in our code (style preference). Not sure how this sneaked in, but let's take this opportunity to fix it:

```suggestion
            raise google.cloud.exceptions.NotFound('blob %s not found in %s' % (key, self._blob.bucket.name))
```